src/commands/login.rs

```diff
@@ -256,10 +264,30 @@ impl LoginCommand {
         let login_connection = match self.auth_method() {
             AuthMethod::Github => self.run_interactive_gh_login().await?,
             AuthMethod::UsernameAndPassword => self.run_interactive_basic_login().await?,
+            AuthMethod::Token => self.login_using_token().await?,
         };
         self.save_login_info(&login_connection)
     }
 
+    async fn login_using_token(&self) -> Result<LoginConnection> {
+        // Validate the token by calling list_apps API until we have a user info API
+        HippoClient::new(ConnectionInfo {
+            url: self.hippo_server_url.to_string(),
+            danger_accept_invalid_certs: self.insecure,
+            api_key: self.token.clone(),
+        })
+        .list_apps()
+        .await
+        .context("token validation failed")?;
```

Comment:
Entered a wrong token and this was the result:

```
$ spin cloud login
Error: token validation failed

Caused by:
```

Reply:
@itowlson earlier suggested:

```
"Login using the provided token failed.  Run spin login or create a new token using the Fermyon Cloud user interface."
```

does this look ok? 

(this will change `token validation failed` with above message. `Caused by:` will remain as is because it seems to be coming from `anyhow` crate)

Reply:
this is how it looks with latest changes:

```
➜  spin login --token abc
Error: Login using the provided token failed. Run `spin login` or create a new token using the Fermyon Cloud user interface.

Caused by:
```